packages/components/src/components/chip/chip.tsx

```diff
@@ -0,0 +1,234 @@
+        previousSelection: previousSelection,
+        currentSelection: []
+      });
+    }
+  }
+
+  handleWrapperClick() {
+    if (!this.readOnly) {
+        this.toggleDropdownMenu();
+     }
+  }
+
+  handleWrapperKeyDown(event: KeyboardEvent) {
+    if (this.readOnly) {
+        if (event.code === 'Space' || event.code === 'Enter') {
+            this.toggleDropdownMenu();
+          }
+    }
+  }
+
+  syncChipState() {
+    const chipItems: NodeList = this.getChipItems();
+    let key: number = 0;
+    chipItems.forEach((chipItem: HTMLIfxChipItemElement) => {
+      chipItem.chipState = {
+        emitIfxChipItemSelect: true,
+        size: (this.size === 'small' ? 'small' : 'large'),
+        variant: (this.variant === 'multi' ? 'multi' : 'single'),
+        key: key++
+      };
+    });
+  }
+
+  componentWillLoad() {
+    /* Propogating the required Chip State with every Chip Item. */
+    this.syncChipState();
+  }
+
+  render() {
+    return (
+      <div aria-value={this.getSelectedOptions()} aria-label='chip with a dropdown menu' class='chip'>
+        <div class={`chip__wrapper chip__wrapper--${this.size === 'small' ? 'small' : 'large'}
+                    chip__wrapper--${this.variant === 'multi' ? 'multi' : 'single'}
+                    ${this.opened && !this.readOnly ? 'chip__wrapper--opened' : ''}
+                    ${this.selectedOptions.length ? 'chip__wrapper--selected' : ''}`}
+          tabIndex={0}
+          onClick={!this.readOnly ? () => { this.handleWrapperClick() } : undefined}
+          onKeyDown={!this.readOnly ? (e) => { this.handleWrapperKeyDown(e) } : undefined}>
+  
+          <div class='wrapper__label'>
+            {
+              (this.selectedOptions.length === 0) && `${this.placeholder}`
+            }
+  
+            {
+              (this.selectedOptions.length !== 0 && this.variant === 'multi') &&
+              `${this.placeholder}:`
+            }
+  
+            {
+              (this.selectedOptions.length !== 0) &&
+              <div class='label__selected-options'>
+                {this.selectedOptions.map(option => option.label).join(', ')}
+              </div>
+            }
+  
+            {/* Number indicator appears only when 2+ options selected in 'multi' variant. */}
+            {
+              (this.selectedOptions.length > 2) &&
+              <ifx-number-indicator>  {`+${this.selectedOptions.length - 2}`} </ifx-number-indicator>
+            }
+          </div>
+  
+          {
+            !this.readOnly && (this.variant !== 'multi' || (this.variant === 'multi' && this.selectedOptions.length === 0)) &&
+            <div class='wrapper__open-button'>
+              <ifx-icon key={1} icon={`chevrondown16`} />
+            </div>
+          }
+  
+          {
+            ((this.selectedOptions.length >= 1) && this.variant === 'multi') &&
+            <div class='wrapper__unselect-button' onClick={(e) => { this.handleUnselectButtonClick(e) }}>
```

Comment:
@verena-ifx 
Bug: Selected options get unselected when `readOnly` is set and cross button is clicked in multi variant.

https://github.com/user-attachments/assets/00ab1da6-123d-4773-ae89-dc7fb2699ca0

Please readOnly prop check for onClick on line 218.